crates/manifest/src/schema/v2.rs

```diff
@@ -95,6 +98,32 @@ impl TryFrom<toml::Value> for ComponentSpec {
     }
 }
 
+/// Component dependency
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(untagged, deny_unknown_fields)]
```

Comment:
Untagged enums are super brittle with strict formats like VersionReq, PackageRef or KebabId.  Like if someone typoes `"a:b" = { package = "c:d", version = "canary" }` they'll get some error about "data did not match any variant of untagged enum ComponentDependency."  (Cf. https://github.com/fermyon/spin/pull/2634.)  I'd strongly encourage being more forgiving at deserialisation time and post-processing into a stricter form when you know which variant you're in!

Reply:
I agree. Just to confirm, should `Registry` and `PackageRef` be specified as `String`s here (and in that PR)?

Reply:
That PR only seems to cover `version`...

Reply:
Updated to relax the constraints in the untagged enum `ComponentDependency`. PTAL (+@ the error messages when parsing in loader), thanks!

Reply:
Yeah the linked PR was just an example of where it bit us most recently.  Anything that can fail to parse is iffy in an untagged enum.

(And since _anything,_ can fail to parse, even ints and strings... but that is another fret for another day.)